File: StykerDG.FarmForge.DataModel/Models/ProductDestination.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text;

namespace StrykerDG.FarmForge.DataModel.Models
{
    [Table("ProductDestination")]
    public class ProductDestination : BaseModel
    {
        [Key]
        public int ProductDestinationId { get; set; }
        public int ProductId { get; set; }
        public int? OrderId { get; set; }
        //[ForeignKey("OrderId")]
        //public Order Order { get; set; }
        public int? CropId { get; set; }
        //[ForeignKey("CropId")]
        //public Crop Crop { get; set; }
        public int? DestinationProductId { get; set; }
        //[ForeignKey("DestinationProductId")]
        //public Product DestinationProduct { get; set; }
    }
}

File: StrykerDG.FarmForge.Client/farmforge_client/lib/models/sales/product_destination.dart
class ProductDestination {
  int productDestinationId;
  int productId;
  int orderId;
  int cropId;
  int destinationProductId;

  ProductDestination({
    this.productDestinationId,
    this.productId,
    this.orderId,
    this.cropId,
    this.destinationProductId
  });

  factory ProductDestination.fromMap(Map<String, dynamic> data) {
    return ProductDestination(
      productDestinationId: data['ProductDestinationId'],
      productId: data['ProductId'],
      orderId: data['OrderId'],
      cropId: data['CropId'],
      destinationProductId: data['DestinationProductId']
    );
  }

  Map<String, dynamic> toMap() {
    return {
      'ProductDestinationId': this.productDestinationId,
      'ProductId': this.productId,
      'OrderId': this.orderId,
      'CropId': this.cropId,
      'DestinationProductId': this.destinationProductId
    };
  }
}